### PR TITLE
feat: assert ts non-neg

### LINKED ISSUE #2686: Return failed if the value of `ts` column is negative
- `Put`/`BulkLoad`  in storage/mem_table.cc and storage/disk_table.cc
- sdk/sql_insert_row.cc

### PR DESCRIPTION
**Pull request by jkpjkpjkp**
* **What kind of change does this PR introduce?** (Bug fix, feature, docs update, ...)
issue#2686

* **What is the current behavior?** (You can also link to an open issue here)

* **What is the new behavior (if this is a feature change)?**

### CODE REVIEW

File: src/codec/codec.cc [hunk 1]
```diff
@@ -346,6 +346,10 @@ bool RowBuilder::SetInt64(int8_t* buf, uint32_t index, int64_t val) {
 }
 
 bool RowBuilder::AppendTimestamp(int64_t val) {
+    if (val < 0) {
+		PDLOG(WARNING, "negative timestamp %d.", val);
```
Comment (dl239) [inline src/codec/codec.cc]: rm log

File: src/storage/mem_table.cc [hunk 2]
```diff
@@ -191,6 +199,8 @@ bool MemTable::Put(uint64_t time, const std::string& value, const Dimensions& di
                     PDLOG(WARNING, "get ts failed. tid %u pid %u", id_, pid_);
                     return false;
                 }
+                if (ts < 0)
```
Comment (dl239) [inline src/storage/mem_table.cc]: add warning log

File: src/storage/disk_table.cc [hunk 3]
```diff
@@ -217,7 +221,11 @@ bool DiskTable::Put(const std::string& pk, uint64_t time, const char* data, uint
 }
 
 bool DiskTable::Put(uint64_t time, const std::string& value, const Dimensions& dimensions) {
-    const int8_t* data = reinterpret_cast<const int8_t*>(value.data());
+    if (time > LLONG_MAX) {
```
Comment (dl239) [inline src/storage/disk_table.cc]: rm all `LLONG_MAX` judgement

File: src/sdk/sql_insert_row.cc [hunk 4]
```diff
@@ -198,6 +198,9 @@ bool SQLInsertRow::AppendInt64(int64_t val) {
 }
 
 bool SQLInsertRow::AppendTimestamp(int64_t val) {
+    if (val < 0) {
+        return false;
+    }
```
Comment (dl239) [inline src/sdk/sql_insert_row.cc:203]: add check in `AppendInt64`

**Comment by dl239** [comment]
add test case

File: src/sdk/sql_insert_row.cc (hunk 4 above)
Comment (jkpjkpjkp) [reply src/sdk/sql_insert_row.cc:203]: should we guarantee `int64_t val` in `AppendInt64` be non-negative?

Comment (dl239) [reply src/sdk/sql_insert_row.cc:203]: guarantee no negative if the col_id in `ts_set_`

File: src/storage/mem_table.cc [hunk 5]
```diff
@@ -148,7 +148,7 @@ bool MemTable::Put(uint64_t time, const std::string& value, const Dimensions& di
         PDLOG(WARNING, "empty dimension. tid %u pid %u", id_, pid_);
         return false;
     }
-    if (value.length() < codec::HEADER_LENGTH) {
+	if (value.length() < codec::HEADER_LENGTH) {
```
Comment (dl239) [inline src/storage/mem_table.cc]: cpplint

File: src/storage/disk_table.cc [hunk 6]
```diff
@@ -248,6 +248,10 @@ bool DiskTable::Put(uint64_t time, const std::string& value, const Dimensions& d
                 PDLOG(WARNING, "get ts failed. tid %u pid %u", id_, pid_);
                 return false;
             }
+            if (ts < 0) {
+                PDLOG(WARNING, "ts < 0 (ts = %d). tid %u pid %u", ts, id_, pid_);
+                continue;
```
Comment (dl239) [inline src/storage/disk_table.cc:253]: return false

File: src/storage/mem_table.cc [hunk 7]
```diff
@@ -191,6 +191,10 @@ bool MemTable::Put(uint64_t time, const std::string& value, const Dimensions& di
                     PDLOG(WARNING, "get ts failed. tid %u pid %u", id_, pid_);
                     return false;
                 }
+                if (ts < 0) {
+                    PDLOG(WARNING, "ts < 0 (ts = %d). tid %u pid %u", ts, id_, pid_);
+                    continue;
```
Comment (dl239) [inline src/storage/mem_table.cc:196]: return false

**Pull request closed without merging**